### ISSUE TITLE
Rust syntax issues

### ISSUE THREAD

**Issue by dten** [sublimehq/Packages#250]
Since changing to the latest dev build my rust does not highlight properly.

```
mod trafile;
mod comment;
mod location;
mod traction;

pub use self::trafile::*;
```

for example the only thing highlighted here is the first `mod`

another stops highlighting after certain macro usages in a macro definition for example

```

pub fn from_buf_reader<T>(s: io::BufReader<T>) -> Result<isize, &'static str>
    where T: io::Read
{

    macro_rules! eat_numbers {
        ($count:expr, $msg:expr) => {{
            let parse_err = concat!("Err parsing value in ", $msg);
            try!{ eat_numbers(&mut lines, $count, parse_err, missing_err, too_many) }
        }}
    };

    let mut starts_stops = eat_numbers!(relief_count_total * 2, "starts and stops");

    let starts = starts_stops.split_off(relief_count_total);
    let stops = starts_stops;

    unimplemented!();
}
```

the highlighting stops at `try!`

I'd try fix this myself but I don't# understand the syntax format :( sad times

**Comment by wbond** [comment]
Thank for you for the code snippets. I should have this fixed up shortly!

**Comment by wbond** [comment]
These have all been fixed in master via the commits from 46918179de52e229628dc9fbcd0603b959ae39c6 to 633f728c50e546714643c6835244ec08544718de.

Please do continue to post issues as you find errors!

**Comment by dten** [comment]
Thanks :)
